### ISSUE TITLE
[Feature] Index mods without author consent and facilitate mandatory manual installation

### ISSUE THREAD

**Issue by politas** [KSP-CKAN/CKAN#2957]
Problem
-------
We have various mod authors de-listed from CKAN due to installation issues. Our [policy](https://github.com/KSP-CKAN/CKAN/blob/master/policy/de-indexing.md) is that authors of FOSS mods can ask for their mods not to be installed by the CKAN client, and we will honour those requests. But it is a problem for users to have to download and track manually installed mod versions separately.

Suggestions
-----------
We need a system to assist users to manually install a mod and have it's metadata tracked by the CKAN client. Ideally, we would have a right-click option on the mod version list which would download the mod version's zip file, and then launch two separate processes - a File system browser window targeted at the KSP instance's GameData folder, and a compression tool window opening the zip file. The user can then look at the instructions in the zip file and copy files wherever they need to go.

The CKAN client should then open a dialog box, asking the user if they successfully installed the mod, which would let us set the mod up as installed, though without any installed files. We'd need a  way to indicate that the mod has been manually installed, but we would be able to use the mod version for dependency purposes.

Once we have this system in place, we can look at adding a new tag to the metadata which would force this system to be used for a given mod, and we could then clarify the de-indexing policy and return to a more inclusive strategy.

**Comment by linuxgurugamer** [comment]
Before moving forward with this, it would be wise to contact some of those authors asking if it would be acceptable.  I can see that some authors are so paranoid that they would freak out again at just the mention of CKAN

**Comment by HebaruSan** [comment]
Agreed, if the goal is to make those modders happy, it would be much better to approach them pro-actively than to have them suddenly receive support requests mentioning that they installed via CKAN using some special new feature, when they thought CKAN installs were banned.

**Comment by linuxgurugamer** [comment]
It's not about keeping them happy, it's to avoid antagonizing them.  We went through a big to-do a few years ago and people have long memories

**Comment by HebaruSan** [comment]
To me the two are synonymous, and yes I remember that blow-up (though I was not yet on this team). Either way, the point would be to treat mod authors such that they feel that their concerns and point of view have been considered, and a little buy-in goes a long way when encountering minor speed bumps with new functionality.

____

This feature would generate some clarity in the community over whether CKAN actually causes install problems. Most of the time the user is going to get two file browser windows and drag one folder to another. Eventually users will ask themselves why they have to do this, when CKAN obviously could do it just as well automatically.

Installation also isn't the only point of contention. Consider FAR, which auto-generates files at run time which CKAN then doesn't remove at uninstallation, which can leave ModuleManager patches depending on FAR active (see #2841). I guess we could also auto-open GameData for the same modules when uninstalling and say, "Uninstall Module X manually" ?

How would this impact CmdLine? Those users may or may not have the ability to open file browser windows. There are plenty of Unix tools that could facilitate a manual installation from the command line (launch a sub shell, suspend process, GNU Screen, etc.), but they're not cross platform.

**Comment by linuxgurugamer** [comment]
One other thing.  Since many mods make new files in their directory, it might be nice to have an option to totally delete the entire mod's directory, with a warning beforehand of course.  
It's necessary because MM looks for the directory being there when deciding if a mod is installed, and an empty directory will still trigger that

**Comment by HebaruSan** [comment]
That's #2841, let's not over-clutter this with comments about that separate issue.

**Comment by linuxgurugamer** [comment]
Just chatted with @Sigma88, he is ok with CKAN providing a link to his mods on Curseforge.
Also, in case  it comes up, he does NOT want any donation links for his mods.  Quoting him: "but I don't want them having donation links for me, not even if the link is a genuine one to my paypal"

**Comment by politas** [comment]
I don't see the need to get buy-in from those modders before we implement the manual installation ability. Before we start adding mods that were opted-out, sure. But I really wanted this to just be a technical change to enable the possibility.

**Comment by HebaruSan** [comment]
If no one consents to being re-added due to this, it would be a lot of wasted effort.

**Comment by HebaruSan** [comment]
Let's not do this. If someone wants it badly enough to work on it and submit a PR, we can talk specifics at that point, but I don't think this solves a problem that we have, and it traduces the notion that CKAN cares about author consent.
